### PR TITLE
CONTRIBUTING.md: extend section on getting your PR merged

### PR DESCRIPTION
**Pull request by pbsds**
I was late to https://github.com/NixOS/nixpkgs/pull/345033, but chance has it that I half a year ago wrote up a draft much like it. @Atemu's text is already excellent, but here I extend it with some of my points. The text is already quite wordy, so i aimed to boil down my changes to the bare minimum

_(btw yes, you can search for `"- [x] Tested, as applicable:"` and `"- [x] Tested basic functionality of all binary files"` ;) )_

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: CONTRIBUTING.md [hunk 1]
```diff
@@ -768,9 +779,10 @@ If someone left an approving review on your PR and didn't merge a few days later
 
 Please see it as your responsibility to actively remind reviewers of your open PRs.
 
-The easiest way to do so is to simply cause them a Github notification.
-Github notifies people involved in the PR when you add a comment to your PR, push your PR or re-request their review.
+The easiest way to do so is to cause them a Github notification.
+Github notifies people involved in the PR whenever you add a comment to your PR, push your PR or re-request their review.
 Doing any of that will get you people's attention again.
+Everyone deserves proper attention, and yes that includes you!
```
Comment (Atemu) [inline CONTRIBUTING.md:785]: Hm, not sure how I feel about this piece of advice. I really do want to get across that committer attention isn't infinite or can in any way be *expected*.

**Review by Atemu — APPROVED** [review]
Changes all LGTM apart from one

File: CONTRIBUTING.md [hunk 2]
```diff
@@ -689,7 +691,9 @@ Ask them nicely whether they still intend to review your PR and find yourself an
 
 ### How can I get a committer to look at my PR?
 
-- Simply wait. Reviewers frequently browse open PRs and may happen to run across yours and take a look.
+- Improve skimmability: use a simple descriptive PR title (details go in commit titles) outlining _what_ is done, possibly also _why_.
```
Comment (fricklerhandwerk) [inline CONTRIBUTING.md:694]:
```suggestion
- Improve skimmability: use a simple descriptive PR title (details go in commit titles) outlining _what_ is done and _why_.
```

The why is the most important thing of any communication that isn't the code itself. If the the diff is huge, a "what" is helpful, and most changes should be small anyway in order to be reviewable.

File: CONTRIBUTING.md [hunk 3]
```diff
@@ -730,6 +734,13 @@ There may be constraints you had to work with which they're not aware of or qual
 
 There are some further pitfalls and realities which this section intends to make you aware of.
 
+### Aim to reduce cycles
+
+Please be prepared for it to take a while before the reviewer gets back to you after you respond.
+This is simply the reality of community projects at the scale of nixpkgs.
```
Comment (fricklerhandwerk) [inline CONTRIBUTING.md:740]:
```suggestion
This is simply the reality of community projects at the scale of Nixpkgs.
```

**Review by fricklerhandwerk — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (pbsds) [reply CONTRIBUTING.md:785]: I myself struggled for a long time to overcome the feeling that I don't deserve to take up any space. Asking for attention is a major blocker for some.

Comment (pbsds) [reply CONTRIBUTING.md:785]: this is also why i prefer to avoid terms like "simply", since it is not that simple for many

Comment (Atemu) [reply CONTRIBUTING.md:785]: I can certainly feel where you're coming from here. Perhaps there's a wording that captures both realities.

**Review by leona-ya — APPROVED** [review]

File: CONTRIBUTING.md (hunk 1 above)
Comment (Atemu) [reply CONTRIBUTING.md:785]:
```suggestion
Everyone deserves proper attention, and yes that includes you! Please be mindful that committers are not always able to give everyone the attention they deserve however.
```

Comment (pbsds) [reply CONTRIBUTING.md:785]: :shipit:

**Review by Atemu — APPROVED** [review]

**Pull request merged**